### PR TITLE
Motherlode mine sack ores overlay

### PR DESCRIPTION
**Pull request by SirGirion**
Originally wanted a way to see how much GP I'd earned in a given mlm session but didn't have time to figure out Java UI to add tooltips to the individual ImageComponents. The style is copied from the blast mine overlay, don't think anyone actually asked for this but I wanted it for myself, figured I'd put it up for a PR.

### CODE REVIEW

**Comment by Nightfirecat** [comment]
How does this differ from #12386?

**Comment by SirGirion** [comment]
This differs because it changes the text overlay for how many of each ore you've gotten to an item overlay which takes up less space. In an ideal world, I would've added tooltips to the ImageComponents with the GE value but that feature isn't built in and I didn't feel like dealing with Java UI development.

**Comment by SirGirion** [comment]
image
@Nightfirecat this is what the overlay would look like

File: runelite-client/src/main/java/net/runelite/client/game/ItemManager.java [hunk 1]
```diff
@@ -460,6 +460,23 @@ public AsyncBufferedImage getImage(int itemId)
 	 * @param quantity
 	 * @return
 	 */
+	public AsyncBufferedImage getImage(int itemId, int quantity)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/game/ItemManager.java:463]: Why is this added? It's not used...

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java [hunk 2]
```diff
@@ -123,4 +123,14 @@ default boolean showOresFound()
 	{
 		return true;
 	}
+
+	@ConfigItem(
+		keyName = "showLootAsItems",
+		name = "Toggle item icons",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java:129]:
```suggestion
		name = "Show ore icons",
```

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java [hunk 3]
```diff
@@ -123,4 +123,14 @@ default boolean showOresFound()
...
+		description = "Shows the gems/ores as their items rather than text"
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java:130]:
```suggestion
		description = "Display collected ores and gems as item images instead of text"
```

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java [hunk 4]
```diff
@@ -123,4 +123,14 @@ default boolean showOresFound()
...
+	)
+	default boolean showLootAsItems()
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java:132]: Change this and the keyName to `showLootIcons()` or something similar

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java [hunk 5]
```diff
@@ -65,60 +71,68 @@ public Dimension render(Graphics2D graphics)
 		int runiteFound = session.getRuniteFound();
 
 		// If no ores have even been collected, don't bother showing anything
-		if (nuggetsFound == 0 && coalFound == 0 && goldFound == 0 && mithrilFound == 0
-			&& adamantiteFound == 0 && runiteFound == 0)
+		// This will statement is true iff all ore counts are 0
+		if ((nuggetsFound | coalFound | goldFound | mithrilFound | adamantiteFound | runiteFound) == 0)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java:75]: This doesn't need to be changed.

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java [hunk 6]
```diff
@@ -65,60 +71,68 @@ public Dimension render(Graphics2D graphics)
...
-				.left("Gold:")
-				.right(Integer.toString(goldFound))
-				.build());
+			panelComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.GOLDEN_NUGGET, nuggetsFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.COAL, coalFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.GOLD_ORE, goldFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.MITHRIL_ORE, mithrilFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.ADAMANTITE_ORE, adamantiteFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.RUNITE_ORE, runiteFound, true)));
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java:88]: This should only display collected ores, and not add images for ores which have 0 quantity.

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/game/ItemManager.java (hunk 1 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/game/ItemManager.java]: I had added it mostly to keep in line w/ how I've always done overloaded methods in Java (which is have the least qualified method call increasingly verbose methods w/ defaults) but there is no actual reason for it so I've removed this method overload.

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java (hunk 2 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java]: Change made

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java (hunk 4 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java]: Change made

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java (hunk 3 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeConfig.java]: Change made

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java (hunk 5 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java]: Yeah was trying to be fancy and save some CPU cycles but grand scheme doesn't matter so will revert back.

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java (hunk 6 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java]: Good call, thought blast furnace showed all ores so that is why I did it this way but I missed the crucial if (count == 0) { continue; } in that overlay.

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java [hunk 7]
```diff
@@ -71,21 +71,39 @@ public Dimension render(Graphics2D graphics)
 		int runiteFound = session.getRuniteFound();
 
 		// If no ores have even been collected, don't bother showing anything
-		// This will statement is true iff all ore counts are 0
-		if ((nuggetsFound | coalFound | goldFound | mithrilFound | adamantiteFound | runiteFound) == 0)
+		if (nuggetsFound == 0 || coalFound == 0 || goldFound == 0 || mithrilFound == 0
+			|| adamantiteFound == 0 || runiteFound == 0)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java:75]: This is not right.

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java (hunk 7 above)
Comment (SirGirion) [reply runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java]: Whoops you're right, made a whoopsie re-typing this.

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java [hunk 8]
```diff
@@ -66,59 +72,85 @@ public Dimension render(Graphics2D graphics)
 
 		// If no ores have even been collected, don't bother showing anything
 		if (nuggetsFound == 0 && coalFound == 0 && goldFound == 0 && mithrilFound == 0
-			&& adamantiteFound == 0 && runiteFound == 0)
+			|| adamantiteFound == 0 && runiteFound == 0)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java:75]: This is still wrong.
```suggestion
			&& adamantiteFound == 0 && runiteFound == 0)
```

**Review by Nightfirecat — CHANGES_REQUESTED** [review]
